packages/@aws-cdk/aws-ecs/lib/images/repository.ts

```diff
@@ -27,7 +34,11 @@ export class RepositoryImage extends ContainerImage {
     super();
   }
 
-  public bind(_scope: Construct, containerDefinition: ContainerDefinition): ContainerImageConfig {
+  public bind(scope: Construct, containerDefinition: ContainerDefinition): ContainerImageConfig {
+    if (this.props.ecrRepository) {
+      scope.node.addWarning("Proper policies need to be attached before pulling from ECR repository.");
```

Comment:
Also suggest to use fromEcrRepository. Probably less cryptic and more actionable